[CN-1083]: Add doc for container resources (#154)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/resource-configuration.adoc

= Configuring Container Resources

:description: pass:q[You can configure resources for Hazelcast, sidecar agent and management center containers.]

{description}

> Detailed documentation about resources can be found link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[in Resource Management for Pods and Containers documentation of Kubernetes].

Reviewer: ```suggestion
Detailed documentation about resources can be found in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.
```

> Detailed documentation about resources can be found in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.

Reviewer 2: ```suggestion
You can find detailed information about resources in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.
```

== Configuring Resources for Hazelcast

> You can configure resources for Hazelcast containers using the `resources` field of the CRD spec.

Reviewer: ```suggestion
You can configure resources for Hazelcast containers using the `resources` field of the CRD specification.
```

Author: `spec` is a field, so it is better to mention as `spec` IMO.

Reviewer: OK, could we say "the CRD `spec` field" just so it is clear? 

Although that is two "fields" in close proximity....are they both called fields? Could we call `resources` a sub-field? My knowledge of K8s terminology is not good. 

Alternatively, just put the ticks around spec (/`spec/`) - that is probably sufficient.

Reviewer 3: > Although that is two "fields" in close proximity....are they both called fields? Could we call resources a sub-field?

`resources` is located under `spec`, so it is a kind of sub-field.

> Alternatively, just put the ticks around the spec.a That is probably sufficient.

Done 👍

.Example configuration
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/hazelcast-resources.yaml[]
----

== Configuring Resources for Sidecar

> You can configure resources for sidecar containers using the `agent.resources` field of the CRD spec.

Reviewer: ```suggestion
You can configure resources for sidecar containers using the `agent.resources` field of the CRD specification.
```

.Example configuration
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/sidecar-resources.yaml[]
----

== Configuring Resources for Management Center

> You can configure resources for Management Center containers using the `resources` field of the CRD spec.

Reviewer: ```suggestion
You can configure resources for Management Center containers using the `resources` field of the CRD specification.
```
